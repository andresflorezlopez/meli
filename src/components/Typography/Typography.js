import React from "react";
import PropTypes from "prop-types";
import {
  StyledParagraph,
  StyledH1,
  StyledH2,
  StyledH3
} from "./Typography.styled";

const renderTypography = props => {
  switch (props.type) {
    case "h1":
      return <StyledH1 {...props} />;
    case "h2":
      return <StyledH2 {...props} />;
    case "h3":
      return <StyledH3 {...props} />;
    default:
      return <StyledParagraph {...props} />;
  }
};

const Typography = props => {
  return renderTypography(props);
};

Typography.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element
  ]).isRequired,
  color: PropTypes.string,
  fontWeight: PropTypes.string,
  textAlign: PropTypes.oneOf(["left", "center", "right", "justify"]),
  type: PropTypes.oneOf(["h1", "h2", "h3", "paragraph"])
};

Typography.defaultProps = {
  color: "primary",
  fontWeight: "normal",
  textAlign: "left",
  type: "paragraph"
};

export default Typography;
