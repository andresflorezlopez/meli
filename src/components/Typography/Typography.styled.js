import styled from "styled-components";

const defaultStyles = props => {
  const { theme, color, textAlign, fontWeight } = props;
  return `
    color: ${theme.colors.fontColor[color]};
    text-align: ${textAlign};
    font-weight: ${fontWeight};
    font-family: ${theme.fontFamily};
  `;
};

export const StyledH1 = styled.h1`
  font-size: ${({ theme }) => theme.fontSize.h1};
  ${props => defaultStyles(props)}
`;
export const StyledH2 = styled.h2`
  font-size: ${({ theme }) => theme.fontSize.h2};
  ${props => defaultStyles(props)}
`;
export const StyledH3 = styled.h3`
  font-size: ${({ theme }) => theme.fontSize.h3};
  ${props => defaultStyles(props)}
`;
export const StyledParagraph = styled.p`
  font-size: ${({ theme }) => theme.fontSize.paragraph};
  ${props => defaultStyles(props)}
`;
