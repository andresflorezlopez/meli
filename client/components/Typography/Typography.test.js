import React from "react";
import Typography from "./Typography";
import { mount } from "enzyme";

import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("Typography", () => {
  test("should render h1 tag", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <Typography type="h1">Example of Typography component</Typography>
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render p tag", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <Typography type="paragraph">
          Example of Typography component
        </Typography>
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
