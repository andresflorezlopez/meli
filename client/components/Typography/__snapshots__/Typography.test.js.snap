// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render h1 tag 1`] = `
<h1
  className="sc-1hvrgbn-0 csixHB"
  color="normal"
  fontWeight="normal"
  type="h1"
>
  Example of Typography component
</h1>
`;

exports[`Typography should render p tag 1`] = `
<p
  className="sc-1hvrgbn-4 hzedTQ"
  color="normal"
  fontWeight="normal"
  type="paragraph"
>
  Example of Typography component
</p>
`;
